pkg/agent/qrm-plugins/network/dynamicpolicy/policy.go

```diff
@@ -0,0 +1,366 @@
+// Copyright 2022 The Katalyst Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dynamicpolicy
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+	pluginapi "k8s.io/kubelet/pkg/apis/resourceplugin/v1alpha1"
+
+	apiconsts "github.com/kubewharf/katalyst-api/pkg/consts"
+	"github.com/kubewharf/katalyst-api/pkg/plugins/skeleton"
+	"github.com/kubewharf/katalyst-core/cmd/katalyst-agent/app/agent"
+	"github.com/kubewharf/katalyst-core/pkg/agent/qrm-plugins/util"
+	"github.com/kubewharf/katalyst-core/pkg/config"
+	"github.com/kubewharf/katalyst-core/pkg/config/generic"
+	"github.com/kubewharf/katalyst-core/pkg/metaserver"
+	"github.com/kubewharf/katalyst-core/pkg/metrics"
+	"github.com/kubewharf/katalyst-core/pkg/util/cgroup/common"
+	cgroupcmutils "github.com/kubewharf/katalyst-core/pkg/util/cgroup/manager"
+	"github.com/kubewharf/katalyst-core/pkg/util/general"
+	"github.com/kubewharf/katalyst-core/pkg/util/native"
+	"github.com/kubewharf/katalyst-core/pkg/util/qos"
+)
+
+const (
+	// NetworkResourcePluginPolicyNameDynamic is the policy name of dynamic network resource plugin
+	NetworkResourcePluginPolicyNameDynamic = "dynamic"
+	// ResourceNameNetwork is the resource name of network
+	ResourceNameNetwork = "network"
+)
+
+// DynamicPolicy is the dynamic network policy
+type DynamicPolicy struct {
+	sync.RWMutex
+
+	name       string
+	stopCh     chan struct{}
+	started    bool
+	qosConfig  *generic.QoSConfiguration
+	emitter    metrics.MetricEmitter
+	metaServer *metaserver.MetaServer
+
+	netClassMap                   map[string]uint32
+	isCgV2Env                     bool
+	applyNetClassFunc             func(podUID, containerId string, data *common.NetClsData) error
+	podLevelNetClsAnnoKey         string
+	podLevelNetAtrributesAnnoKeys []string
```

Comment:
typo：podLevelNetAtrributesAnnoKeys -> Attributes, and

Reply:
fixed